### ISSUE TITLE
Deliver draggable row re-ordering enhancement

### ISSUE THREAD

**Issue by enzocol** [carbon-design-system/ibm-products#8818]
This is a related follow on to the item [#80](https://github.com/carbon-design-system/tanstack-carbon/issues/80)

### Resources
[Design file](https://www.figma.com/design/2UKyOU3C9ALlRhcsptSvZI/Data-table---Row-re-ordering?node-id=95-19284&t=vmMl4xzUb37DFZTe-11)

## Tasks 

- [ ] Ensure accessibility by reviewing with @jjennings7 and the accessibility team 
- [ ] Review feasibility and dev implementation with @amal-k-joy 
- [ ] Spec out the solution
- [ ] Discuss as a team

**Comment by enzocol** [comment]
### Feedback on Accessibility
1. **Jess** suggests the overflow menu and reorder mode options may be better for accessibility and usability than the move icon. 
2. The move action should be in the overflow menu if the move icon is needed for technical reasons (skip the focus on icon)

**Comment by enzocol** [comment]
Follow on item [#85](https://github.com/carbon-design-system/tanstack-carbon/issues/85)
